include/revng-c/mlir/Dialect/Clift/IR/CliftAttributes.td

```diff
@@ -15,7 +15,7 @@ class Clift_Attr<string name, string attrMnemonic, list<Trait> traits = []>
   let mnemonic = attrMnemonic;
 }
 
-def FunctionArgumentAttr: Clift_Attr<"FunctionArgument", "farg", [DeclareAttrInterfaceMethods<SubElementAttrInterface>]> {
```

Comment:
Brief recap of why this is necessary:
- you tried removing the `Clift_` prefix from everywhere, and doing that doesn't work because the un-prefixed names clash with other auto-generated stuff we can't control
- so we decided to apply the prefix to everything for consistency

One question though: can the prefix be just `Clift` instead of `Clift_`? `Clift_` is ugly, because it's not `CamelCase`, nor `snake_case`. I'd be much happier if we can make it `CamelCase` like everything else.

Reply:
I don't know if there is a way to configure this, but by default it removes the `Clift_` prefix from the generated class names, but not just `Clift`.

Reply:
https://mlir.llvm.org/docs/DefiningDialects/Operations/#class-name-and-namespaces

Reply:
https://github.com/revng/llvm-project/blob/master/mlir/lib/TableGen/Operator.cpp#L41

Reply:
So `Clift_FunctionArgumentAttr` causes TableGen to generate `clift::FunctionArgumentAttr` in c++, right?

If so, it makes sense, and feel free to close this thread.

Reply:
That's my understanding of it, yes.